frontend/src/app/versions.tsx

```diff
@@ -419,29 +432,36 @@ class Versions extends Component<IProps, any> {
     }
 
     private changePublishState = (buttonText) => {
-        const formData = new FormData();
-        if (buttonText === "Publish") {
-            formData.append(":operation", "pant:release");
-            // console.log('Published file path:', this.props.modulePath)
-            this.draft[0].version = "";
+        // Validate productValue before Publishs
+        if (this.props.productInfo !== undefined && this.props.productInfo.trim() === "" && buttonText === "Publish") {
+            this.setState({ canChangePublishState: false, publishAlertVisible: true })
         } else {
-            formData.append(":operation", "pant:unpublish");
-            // console.log('Unpublished file path:', this.props.modulePath);
-            this.release[0].version = "";
-        }
-        fetch("/content" + this.props.modulePath, {
-            body: formData,
-            method: 'post'
-        }).then(response => {
-            if (response.status === 201 || response.status === 200) {
-                // console.log(buttonText + " works: " + response.status)
-                this.setState({ initialLoad: true, changePublishState: true })
-            } else {
-                // console.log(buttonText + " failed " + response.status)
-                this.setState({ initialLoad: true, changePublishState: true })
-            }
-        });
 
+            if (this.state.canChangePublishState === true) {
+                const formData = new FormData();
+                if (buttonText === "Publish") {
+                    formData.append(":operation", "pant:release");
+                    // console.log('Published file path:', this.props.modulePath)
+                    this.draft[0].version = "";
+                } else {
+                    formData.append(":operation", "pant:unpublish");
+                    // console.log('Unpublished file path:', this.props.modulePath);
+                    this.release[0].version = "";
+                }
+                fetch("/content" + this.props.modulePath, {
+                    body: formData,
+                    method: 'post'
+                }).then(response => {
+                    if (response.status === 201 || response.status === 200) {
+                        console.log(buttonText + " works: " + response.status)
+                        this.setState({ initialLoad: true, changePublishState: true, publishAlertVisble: false, canChangePublishState: true })
+                    } else {
+                        console.log(buttonText + " failed " + response.status)
+                        this.setState({ initialLoad: true, changePublishState: true, publishAlertVisble: true })
```

Comment:
These references to `publishAlertVisible` and `changePublishState` don't appear to be necessary since neither field is used anywhere else in the file.

Reply:
publishAlertVisible is used to show the alert message when medata data is not updated before publishing. 

I removed "changePublishState"

Reply:
Ah... it appears you have a typo here then. The `publishAlertVisble` that you have here is missing an 'i' in the field name. Elsewhere in the code, there is no typo, so this is technically a different field. Subsequently, this code probably does not do what you think it does.

Reply:
the typo has been corrected.